app/controllers/posts_controller.rb

```diff
@@ -1,9 +1,7 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    user_id = params[:user_id]
-    @posts = Post.where(user_id:)
-    @comments = Comment.all
+    @posts = @user.posts
```

Comment:
- [ ] It would be better if we fix the `N+1` query problem when fetching for all post and comments for a user. Although you have created a method, you have not used the method in the code. Here is how we can fix it. Instead of `@posts = @user.posts` use `@posts = @user.posts.includes(:comments)`. Once this is done, kindly take a screenshot of the before and after fixing the `n+1` query problem as we have currently in the PR

Reply:
@addod19 
Thank you so much for the recommendation, we have made the change at this [commit](https://github.com/Lucash2022/Blog_App/pull/11/commits/409eb10ceecb94744e843eb1d2c7abaae47c0ca4)